### ISSUE TITLE
Window does not resize when being dragged between displays with different DPI

### ISSUE THREAD

**Issue by Balkoth** [microsoft/WindowsAppSDK#1972]
### Describe the bug

The main application window does not resize itself when being dragged between displays with different DPI.

Primary is a 4K 150% DPI display.
Secondary is a FHD 100% DPI display.

### Steps to reproduce the bug

1.) Create a new WinUI 3 app with the template: Blank App, Packaged (WinUI 3 in Desktop)
2.) Build an run the app
3.) Move the app between two displays with differing DPI settings

### Expected behavior

The main app window should resize.

### Screenshots

The main app window stays the same on all displays.

### NuGet package version

1.0.0

### Packaging type

Packaged (MSIX), Unpackaged

### Windows version

Windows 10 version 2004 (19041, May 2020 Update)

### IDE

Visual Studio 2022

**Comment by nuuSolutions** [comment]
Interesting. Need this, too. (And I don't even know how to set a dpi (in)dependent size, yet.)
Probably one has to listen to some (moved/switchedscreen/whatever) event to get the info and then react accordingly. Just guessing.

**Comment by WelterDevelopment** [comment]
I don't have a second screen right now, but when I change the scale on the fly in Win11 with a WinAppSDK 1.0 app opened, the app **does** resize itself. If it really is the case that the app does not resize when dragged to a screen with different DPI, I guess you will need to write it by yourself; something like this:
``` xaml
<Grid x:Name="RootGrid" Loaded="RootGrid_Loaded" >
  <TextBlock x:Name="Log"></TextBlock>
</Grid>
```
``` csharp
public AppWindow AW { get; set; }

private AppWindow GetAppWindowForCurrentWindow()
{
	IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
	WindowId myWndId = Win32Interop.GetWindowIdFromWindow(hWnd);
	return AppWindow.GetFromWindowId(myWndId);
}

public MainWindow()
{
	this.InitializeComponent();
	AW = GetAppWindowForCurrentWindow();
	AW.Title = "App title goes here";
}

private double RasterizationScale;
private SizeInt32 AppSize;
private void RootGrid_Loaded(object sender, RoutedEventArgs e)
{
	RasterizationScale = RootGrid.XamlRoot.RasterizationScale;
	AppSize = AW.Size;
	Log.Text = $"Starting Scale: {RasterizationScale:0.00}";

	RootGrid.XamlRoot.Changed += (a, b) =>
	{
		if (RootGrid.XamlRoot.RasterizationScale != RasterizationScale)
		{
			Log.Text = $"Scale changed from {RasterizationScale:0.00} to { RootGrid.XamlRoot.RasterizationScale:0.00}";
			double scaleRatio = RootGrid.XamlRoot.RasterizationScale / RasterizationScale;
			SizeInt32 newSize = new((int)(AppSize.Width * scaleRatio), (int)(AppSize.Height * scaleRatio));
			AW.Resize(newSize);

			RasterizationScale = RootGrid.XamlRoot.RasterizationScale;
			AppSize = newSize;
		}
	};
}
```
I am not sure if the calculation of the scaleRatio is correct; maybe it's more complicated.

**Comment by Balkoth** [comment]
Doing this yourself can only be a workaround until this is fixed in the framework.

**Comment by WelterDevelopment** [comment]
Yep but we are all stuck with the current set of capabilities of WinAppSDK and the next version will come in Q2, so there are 3 months of wait for anything to be implemented production-ready... Until then my workaround might help ;)

**Comment by Balkoth** [comment]
I don't get why this is even bugged. Works without problems in UWP.

**Comment by rkarman** [comment]
Hey @marb2000, I missed this while I was OOF during Jan... This had been assigned to the windowing area, but this is actually a WinUI issue, can you help route this and potentially create a corresponding tracking bug in the WinUI repository? Thanks.
